stream_alert/rule_processor/rules_engine.py

```diff
@@ -443,14 +439,20 @@ def rule_analysis(record, rule, payload, alerts):
             LOGGER.info('Rule [%s] triggered alert [%s] on log type [%s] from entity \'%s\' '
                         'in service \'%s\'', rule.rule_name, alert_id, payload.log_source,
                         payload.entity, payload.service())
+
+            # Combine the required alert outputs with the ones for this rule
+            all_outputs = self._required_outputs
+            if rule.outputs:
+                all_outputs = all_outputs.union(set(rule.outputs))
+
             alert = {
                 'id': alert_id,
                 'record': record,
                 'rule_name': rule.rule_name,
                 'rule_description': rule.rule_function.__doc__ or DEFAULT_RULE_DESCRIPTION,
                 'log_source': str(payload.log_source),
                 'log_type': payload.type,
-                'outputs': rule.outputs,
+                'outputs': list(all_outputs),
```

Comment:
We actually are [saving the outputs as a `StringSet` in Dynamo](https://github.com/airbnb/streamalert/blob/e0839db8d79944f2568c1137bdd206995b88394b/stream_alert/rule_processor/alert_forward.py#L53), so I'd say let's keep this as a set

Reply:
adding a TODO to change this :)

Reply:
Yeah, you still need JSON-compatibility for the alert processor (but not for much longer!)